server/WFCORE-4291_restore_legacy_not_graceful_startup_mode.adoc

```diff
@@ -0,0 +1,69 @@
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Since Wildfly 11, the server starts up in suspended mode, thus rejecting all incoming requests, until the server has completed booting, at which point the state is set to `RUNNING` and requests are accepted. While this helps ensure that the system is in a stable state before allowing any work to happen, it does have a negative effect on some deployment scenarios reported by Wildfly users. 
+
+One example scenario is this: if Deployment A depends -- at startup -- on external configuration, fetched by an HTTP request to Deployment B, Deployment A will fail to start as the HTTP request for that configuration will be rejected by the starting server.
+
+This RFE proposes a new startup mode, `graceless`, that will allow -- for some subsystems, at least -- requests to be handled much earlier in the application server boot process. In standalone mode, the system administrator can choose to enable this startup mode for that server, and in domain mode, the server group can be configured similarly, applying the configuration choice to all servers in that domain.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4291[WFCORE-4291]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1320[EAP7-1320]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+* [ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+For standalone mode, this option will be exposed as a command line option only, with the addition of a `--graceless` option. Since the value is needed before standalone.xml is parsed, it must come from the command line. Given this lack of start mode persistence, system administrators managing standalone instances will need to update their system scripts (`/etc/init.d`, `sysvinit`, `systemd`, etc.)
+
+For domain mode, the configuration will be extended to allow the system administrator to configure how the Host Contoller will start the servers in the group. Every server in the group will be configured to start in the same manner.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+* Persistent standalone graceless configuration
+* Existing system scripts will not be modified as usage of this new flag is covered already by `JBOSS_OPTS`.
+
+== Test Plan
+
+TBD
+
+== Community Documentation
+
+* Describe new --graceless CLI param in https://docs.wildfly.org/21/Admin_Guide#other-command-line-parameters[section 11.1.2]
+* Add a new section describing graceless startup to https://docs.wildfly.org/21/Admin_Guide#Suspend[section 11.2]
+
```

Comment:
IIRC the Undertow subsystem's host resource's 'queue-requests-on-start' attribute will have subtly different behavior with '--graceless'; i.e. if not defined then queueing will be turned off, despite the default value of 'true'.  The free form text description of the attribute should discuss this.

I mention this here because those free form text descriptions of the management API are a form of "community documentation".